### ISSUE TITLE
Upgrade to 1.1.8 fails

### ISSUE THREAD

**Issue by xaloc33** [YunoHost-Apps/pleroma_ynh#102]
When trying to upgrade to 1.1.8 with ```yunohost app upgrade pleroma -u https://github.com/YunoHost-Apps/pleroma_ynh``` the upgrade fails with the following log:

https://paste.yunohost.org/raw/mukoliwasu

Previous version 1.1.7 from testing.

**Comment by yalh76** [comment]
Can you retry to upgrade to be sure it's not a one time error ?

**Comment by xaloc33** [comment]
Already done it :relaxed:. I tried yesterday, retried today and also tried from the testing branch. Same error every time.

**Comment by yalh76** [comment]
> 
> 
> Already done it 🙂. I tried yesterday, retried today and also tried from the testing branch. Same error every time.
> 

Well doing additional tests to check what happens during upgrade

> Edit: I think that before the error it says that the pleroma service is started instead of stopped, but I don't know why since the commands seem to try to stop it.
That's fully normal => related to https://github.com/YunoHost/yunohost/pull/867

**Comment by xaloc33** [comment]
Here the new logs trying the update from testing, still similar error.

https://paste.yunohost.org/raw/axucukijos

**Comment by xaloc33** [comment]
Well, trying stuff I broke the pleroma service, after that when I tried the upgrade it worked. The problem is that the log is too long to share it with yunohost log display --share...

Also ```pleroma_ctl migrate``` doesn't seem to work, it only says that it is: Already up. I have already an open issue about that [here](https://git.pleroma.social/pleroma/pleroma-support/issues/24)

**Comment by yalh76** [comment]
> 
> 
> Here the new logs trying the update from testing, still similar error.
> 
> https://paste.yunohost.org/raw/axucukijos

I've updated the upgrade script on testing branch, currently testing it

**Comment by yalh76** [comment]
> 
> 
> Well, trying stuff I broke the pleroma service, after that when I tried the upgrade it worked. The problem is that the log is too long to share it with yunohost log display --share...
> 
> Also `pleroma_ctl migrate` doesn't seem to work, it only says that it is: Already up. I have already an open issue about that [here](https://git.pleroma.social/pleroma/pleroma-support/issues/24)

Maybe the errors you have are related to an improper upgrade to 1.1.8

**Comment by xaloc33** [comment]
> I've updated the upgrade script on testing branch, currently testing it

 I can try it once it is ready if needed

> Maybe the errors you have are related to an improper upgrade to 1.1.8

Should I retry the update or wait for the fix in the testing branch?


And thanks again for the work and sorry for all the trouble

**Comment by yalh76** [comment]
The point is, upgrade has to be done from 1.1.7. If you are at 1.1.8, you have to remove and restore from backup, but first you have to be sure you have a backup of 1.1.7

**Comment by xaloc33** [comment]
> The point is, upgrade has to be done from 1.1.7. If you are at 1.1.8, you have to remove and restore from backup, but first you have to be sure you have a backup of 1.1.7

Done from which branch should I upgrade?

**Comment by yalh76** [comment]
You can try to upgrade from the testing branch

**Comment by xaloc33** [comment]
> You can try to upgrade from the testing branch

Done, update was successful, but still the same issues with migrate...

**Comment by yalh76** [comment]
From my tests, when upgrading from 1.1.7 to 1.1.8, result for `pleroma_ctl migrate` is `[info] Already up`. Seems ok to me as `pleroma_ctl migrate` is to update database and each pleroma release doesn't require to update the database...

**Comment by xaloc33** [comment]
At some point one of the failed upgrades messed up the database, there was only one user in users. 
I have restored the database and rerun the migrations. I have the same errors that I had before, that are on the [issue](https://git.pleroma.social/pleroma/pleroma-support/issues/24) I opened in pleroma support.

I'm not sure if it is a yunohost related problem or if it is due to pleroma.

Anyway thanks for the help

**Comment by xaloc33** [comment]
Since I couldn't solve the problem I tried a fresh install. The fresh install was failing to due to the fact that there were spaces in my instance name. So the update problem might be related. I've reinstalled pleroma now, restored a previous database and run the migrations. So my instance is up again :smiley: .

Thanks for helping
